pkg/reconciler/v1alpha1/taskrun/resources/cloudevent/cloudevent.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudevent
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/cloudevents/sdk-go/pkg/cloudevents"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/client"
+	cecontext "github.com/cloudevents/sdk-go/pkg/cloudevents/context"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/types"
+	"github.com/knative/eventing-sources/pkg/kncloudevents"
+	"github.com/knative/pkg/apis"
+	"go.uber.org/zap"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+)
+
+// TektonEventType holds the types of cloud events sent by Tekton
+type TektonEventType string
+
+const (
+	// TektonTaskRunUnknown is sent for TaskRuns with "ConditionSucceeded" "Unknown"
+	TektonTaskRunUnknown TektonEventType = "dev.tekton.event.task.unknown"
```

Comment:
Would you add the version to these types?

> Type: String
> Description: This attribute contains a value describing the type of event related to the originating occurrence. Often this attribute is used for routing, observability, policy enforcement, etc. The format of this is producer defined and might include information such as the version of the type - see Versioning of Attributes in the Primer for more information https://github.com/cloudevents/spec/blob/master/primer.md#versioning-of-attributes

Reply:
Yeah, the event types become part of the interface, so we'll need a way to version them.
I'll address this in the dedicated PR

Reply:
https://github.com/tektoncd/pipeline/pull/1112